plugins/action/iptables_state.py

```diff
@@ -81,13 +84,16 @@ def _async_result(self, async_status_args, task_vars, timeout):
 
         return async_result
 
-    def run(self, tmp=None, task_vars=None):
+    def run(self, tmp: str | None = None, task_vars: dict[str, t.Any] | None = None) -> dict[str, t.Any]:
         self._supports_check_mode = True
         self._supports_async = True
 
         result = super().run(tmp, task_vars)
         del tmp  # tmp no longer has any effect
 
+        if task_vars is None:
+            task_vars = {}
```

Comment:
This should also be list as a bugfix

Reply:
If it does not make any difference indeed, I would suggest moving it up. Otherwise the unaware reader one year from now will be asking themselves the same question again: "why are we using the value of the param _before_ validating/normalising it?"